packages/desktop-gui/src/specs/specs-list.jsx

```diff
@@ -40,31 +43,12 @@ class SpecsList extends Component {
     }
 
     const areTestsRunning = this._areTestsRunning()
-    let runSpecsLabel = allSpecsSpec.displayName
-    let runButtonDisabled = false
 
-    if (areTestsRunning && this.runAllSavedLabel) {
-      runSpecsLabel = this.runAllSavedLabel
-    } else {
-      if (hasSpecFilter) {
-        if (numberOfShownSpecs < 1) {
-          runSpecsLabel = 'No specs'
-          runButtonDisabled = true
-        } else {
-          const specLabel = numberOfShownSpecs === 1 ? 'spec' : 'specs'
-
-          runSpecsLabel = `Run ${numberOfShownSpecs} ${specLabel}`
-        }
-      }
-    }
+    // store in the component for ease of sharing with other methods
+    const runWord = areTestsRunning ? 'Running' : 'Run'
```

Comment:
Seems like you could pull this and the labels into their own reusable methods, cause you're repeating the 'Running' : 'Run' check + label constructions here in the render method and also in the `_selectSpec`